### PR TITLE
NIH Harvester & Normalizer

### PR DESCRIPTION
**Pull request by caseyrollins**
@chrisseto for you 🌹

### CODE REVIEW

File: providers/gov/nih/normalizer.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        budget_start = ctx.row.BUDGET_START
+        budget_end = ctx.row.BUDGET_END
+        cfda_code = ctx.row.CFDA_CODE
+        core_project_number = ctx.row.CORE_PROJECT_NUM
+        ed_inst_type = ctx.row.ED_INST_TYPE
+        fiscal_year = ctx.row.FY
+        foa_number = ctx.row.FOA_NUMBER
+        full_project_number = ctx.row.FULL_PROJECT_NUM
+        nih_spending_cats = ctx.row.NIH_SPENDING_CATS
+        organization_city = ctx.row.ORG_CITY
```
Comment (chrisseto) [inline providers/gov/nih/normalizer.py]: Theses should go on the organization

File: providers/gov/nih/normalizer.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    class Extra:
+        arra_funded = ctx.ARRA_FUNDED
+        award_notice_date = ctx.AWARD_NOTICE_DATE
+
+
+class ThroughAwards(Parser):
+    award = Delegate(Award, ctx)
+
+
+class CreativeWork(Parser):
```
Comment (chrisseto) [inline providers/gov/nih/normalizer.py]: Looks like this should maybe be a project?

Comment (caseyrollins) [inline providers/gov/nih/normalizer.py]: Good call, yes it should be.

**Comment by caseyrollins** [comment]
@chrisseto 🔙  to you

File: share/normalize/utils.py [hunk 3]
```diff
@@ -34,13 +34,13 @@ def format_address(self, address1='', address2='', city='', state_or_province=''
     elif address1 and address2:
         return '{}\n{}'.format(address1, address2)
 
-    elif city and state_or_province:
-        return '{}, {}'.format(city, state_or_province)
+    elif city and state_or_province and postal_code and country:
```
Comment (chrisseto) [inline share/normalize/utils.py]: https://github.com/CenterForOpenScience/SHARE/pull/218#discussion_r71175865

File: providers/gov/nih/normalizer.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+class Association(Parser):
+    pass
+
+
+class Project(Parser):
+
+    PROJECT_BASE_URL = 'https://projectreporter.nih.gov/project_info_description.cfm?aid={}'
+    FOA_BASE_URL = 'https://grants.nih.gov/grants/guide/pa-files/{}.html'
+
+    def format_nih_url(self, id):
```
Comment (chrisseto) [inline providers/gov/nih/normalizer.py]: https://github.com/CenterForOpenScience/SHARE/pull/218#discussion_r71533780

Note: the `Extra` class is also considered a class level variable but should go after other class level variables.

**Comment by caseyrollins** [comment]
@chrisseto 🌜

**Comment by chrisseto** [comment]
Merged via https://github.com/CenterForOpenScience/SHARE/commit/dce88de4a7cedd002e2a9d3823c7729b9615a100

**Pull request closed without merging**